server/handlers/result_handlers.py

```diff
@@ -175,9 +179,27 @@ def client(self) -> LSEClient:
     @model_validator(mode="after")
     def ready(self):
         # Use versions endpoint to verify connection to LS instance
-        assert self.client.versions.get()
-
-        return self
+        # First attempt without retry to quickly catch auth/config issues
+        try:
+            self.client.versions.get()
+            logger.info(f"LSE client connection verified")
+            return self
+        except Exception as e:
+            # Check if this is a rate limit that should be retried
```

Comment:
lets leaver TODO: to remove this part as we make more robust inter-service communication